src/main/java/com/example/controller/GraphController.java

```diff
@@ -109,19 +111,13 @@ private void setGraphViewBindings(){
             // bind vertex traitor property with vertex color
             MyVertex<Integer> vertex = (MyVertex<Integer>)graphVertex.getUnderlyingVertex();
             vertex.getIsTraitor().addListener(changed -> {
```

Comment:
Something weird happens when we generate random traitors and later we change their traitors flag. Please check this out and if it turns out to be complex problem, add Jira ticket for it.

Reply:
Fixed. I'm not sure why but the logic we had before didn't work in some cases so I added a method which adds listeners for newly created vertices. However, we have to remember to call that method for each vertex (refers to generating graphs or reading them from files).